sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/model/PartitionMetadata.java

```diff
@@ -188,6 +217,20 @@ public Timestamp getCreatedAt() {
     return finishedAt;
   }
 
+  /**
+   * Estimates the size in bytes of a partition when encoded using Avro.
+   *
+   * @return the number of bytes of the encoded object
+   */
+  public long bytesSize() {
```

Comment:
Does this mean we have to encode the Data change record everytime we call this function? Would that be expensive?

Reply:
Yes, but it is not so expensive, see the results above from the benchmark. At p100 `AvroCoder` beats the previous strategy